### PR TITLE
Support --use-core and --ctypes-prefix

### PR DESCRIPTION
**Pull request by emilio**
As requested in https://github.com/Yamakaky/rust-bindgen/issues/400#issuecomment-258632805.

r? @fitzgen

### CODE REVIEW

File: tests/headers/no-std.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// bindgen-flags: --ctypes-prefix "libc" --use-core --raw-line "#![no_std]" --raw-line "mod libc { pub type c_int = i32; pub enum c_void {} }"
```
Comment (emilio) [inline tests/headers/no-std.h]: poor man's libc :)

File: tests/headers/use-core.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// bindgen-flags: --use-core --raw-line "extern crate core;"
```
Comment (fitzgen) [inline tests/headers/use-core.h]: Shouldn't `--use-core` imply `extern crate core;` in the bindings? Shouldn't we automatically insert the extern crate definition?

**Comment by cpavlina** [comment]
Wow, that was fast. Thank you!

File: tests/headers/use-core.h (hunk 2 above)
Comment (emilio) [inline tests/headers/use-core.h]: Nope, at least not completely automatically I guess.

It's not uncommon doing `include!("path/to/bindings.rs")`, in which case it's unnecessary. In general it's completely unnecessary as long as the bindings aren't the crate root, so I went the easy/safe way. if it's harmless we can do it in a followup, in which case is straight-forward to do.

**Comment by jdub** [comment]
Doesn't seem to output `core` in every instance, e.g. this is the output of the `func_ptr.h` test:

```
/* automatically generated by rust-bindgen */

extern "C" {
    #[link_name = "foo"]
    pub static mut foo:
               ::std::option::Option<unsafe extern "C" fn(x:
                                                              ::std::os::raw::c_int,
                                                          y:
                                                              ::std::os::raw::c_int)
                                         -> ::std::os::raw::c_int>;
}
```

**Comment by emilio** [comment]
@jdub good catch, that's a one-off :)

https://github.com/servo/rust-bindgen/pull/229

**Pull request merged**